test/forms/declaration/AdditionalFiscalReferenceSpec.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.declaration
+
+import org.scalatest.{MustMatchers, WordSpec}
+import play.api.libs.json.{JsObject, JsString, JsValue}
+
+class AdditionalFiscalReferenceSpec extends WordSpec with MustMatchers {
+  import AdditionalFiscalReferenceSpec._
+
+  "Bound form with AdditionalFiscalReference mapping" should {
+
+    "not contain errors" when {
+      "form is correct" in {
+        val form = AdditionalFiscalReference.form().bind(correctForm)
+
+        form.errors must be(empty)
+      }
+    }
+    "contain errors" when {
+      "country is incorrect" in {
+        val form = AdditionalFiscalReference.form().bind(incorrectCountry)
+
+        form.errors.length must be(1)
+        form.errors.head.message must be("declaration.additionalFiscalReferences.country.error")
+      }
+
+      "reference is incorrect" in {
+        val form = AdditionalFiscalReference.form().bind(incorrectReference)
+
+        form.errors.length must be(1)
+        form.errors.head.message must be("declaration.additionalFiscalReferences.reference.error")
+      }
+      "both country and reference are incorrect" in {
+        val form = AdditionalFiscalReference.form().bind(incorrectCountryAndRef)
+
+        form.errors.length must be(2)
```

Comment:
could map over the list and assert that way - would make 3 assertions into 1

Reply:
I don't know how. We need to check different messages inside the error.

Can you show some example?

Reply:
Changed